client/src/pages/settings-recipe/settings-recipe.ts

```diff
@@ -41,4 +44,16 @@ export class SettingsRecipePage {
     this.nav.pop();
   }
 
+  addIngredient() {
+    let addIngredientModal = this.modalCtrl.create(AddIngredientPage, {recipeTitle: this.recipe.name});
+    addIngredientModal.onDidDismiss(ingredient => {
+      if(ingredient) this.recipe.ingredients.push({id: ingredient.id, amount: 40});
```

Comment:
Moving onDidDismiss handler logic to class method and calling it from arrow function would be cleaner. Also hardcoded amount (40) could be changed to class attribute defaultNewIngredientAmount or something like that.